libraries/logging/stream-logger/src/main/java/io/getstream/logging/StreamLog.kt

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.logging
+
+import io.getstream.logging.StreamLogger.Level.DEBUG
+import io.getstream.logging.StreamLogger.Level.ERROR
+import io.getstream.logging.StreamLogger.Level.INFO
+import io.getstream.logging.StreamLogger.Level.VERBOSE
+import io.getstream.logging.StreamLogger.Level.WARN
+
+public object StreamLog {
+
+    @PublishedApi
+    internal var logger: StreamLogger = SilentStreamLogger
+    @PublishedApi
+    internal var validator: IsLoggableValidator = IsLoggableValidator { _, _ -> false }
+
+    /**
+     * Sets custom [StreamLogger] implementation.
+     */
+    public fun init(
```

Comment:
It doesn't seem an `init()` but a `set()` method.
An `init()` method is expected to be called at the beginning and only once, and if by some reason it is called twice, notify about it.
WDYT about to rename it?

Reply:
On the other hand, we we need to update both params in one shot?
Maybe we can only use `var` fields and allow to update it from outside